File: 0131-palindrome-partitioning/0131-palindrome-partitioning.cpp
class Solution {
public:
bool ispalindrome(string s)
{
    int len=s.length();
    int i=0,j=len-1;
   
   while(i<=j)
   {
    if(s[i]!=s[j])
    return false;
    i++;
    j--;
   }
   return true;
}
vector<vector<string>>ans;
void helper(int ind,string &s,vector<string >&temp)
{
    if(ind==s.length())
    {
        for(int i=0;i<temp.size();i++)
        {
            if(!ispalindrome(temp[i]))
            return;
        }
        ans.push_back(temp); 
        return;
    }

    for(int j=ind;j<s.length();j++)
    {
        temp.push_back(s.substr(ind,j-ind+1));
        helper(j+1,s,temp);
        temp.pop_back();
    }
   return;
}
    vector<vector<string>> partition(string s) {
        
        vector<string>temp;
        helper(0,s,temp);
        return ans;
    }
};

File: 0131-palindrome-partitioning/0131-palindrome-partitioning.java
class Solution {
    
    private boolean isPalindrome(String s,int i,int j)
    {
        while(i<j)
        {
            if(s.charAt(i)!=s.charAt(j))
            return false;
            i++;
            j--;
        }
        return true;
    }

    private void generate(int ind,String s,List<String>l,List<List<String>>ans)
    {
         
         if(ind==s.length())
         {
            ans.add(new ArrayList<>(l));
            return;
         }

         for(int i=ind;i<s.length();i++)
         {
             if(isPalindrome(s,ind,i))
             {
                String t=s.substring(ind,i+1);

                l.add(t);
                generate(i+1,s,l,ans);
                l.remove(l.size()-1);
             }
         }
    }
    public List<List<String>> partition(String s) {
        
        List<List<String>>ans=new ArrayList<>();
        List<String>t=new ArrayList<>();

        generate(0,s,t,ans);
        return ans;
          
            }
}